spec/system/page_objects/pages/admin.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module PageObjects
+  module Pages
+    class Admin < PageObjects::Pages::Base
+      def visit_filtered_plugin_setting(filter)
+        visit("/admin/site_settings/category/plugins?filter=#{filter}")
+        self
+      end
+
+      def toggle_setting(text)
```

Comment:
I took a look at how this is used in the other PR, and I think instead of relying on hardcoded setting text we should instead just introduce a data-attr or a class onto the `.row.setting` div that is the actual setting name, so we can target it in a more unique and succinct way:

![image](https://user-images.githubusercontent.com/920448/201229216-2065d94f-7f4d-4582-b8d5-10e91bf6cdc6.png)

Then this would become:

```ruby
toggle_setting("voting_enabled")

def toggle_setting(setting_name)
  find('.admin-detail .row.setting[data-setting="#{setting_name}"]').find('.setting-value span').click
end
```

Then we could use the same for other setting changes that aren't toggling (e.g. if we need to change a setting dropdown at some point)

I also perhaps think that `AdminSettings` would be a better name for this page object.

Reply:
> toggle_setting("voting_enabled")

Yeah this is nice and matches the SiteSetting. I was thinking initially of the text to do a more "e2e"/"integrate-y" style test which would test i18n keys resolved correctly. Most recently I merged a change which didn't update the key and resulted in a bug. 😉 

> AdminSettings would be a better name

I initially named this one AdminSettings, weirdly I don't remember now why I changed it... 🧐

Reply:
I see what you are going for, however I would prefer just using the setting since it is more succinct. I think if we want to test the test specifically inside the system specs, perhaps we should use the I18n helper in the test e.g. `expect(page).to have_content(I18n.t("discourse_upvotes.voting_enabled"))` (or whatever the particular key is). That way the interactions with the settings themselves are just driver/autopilot specific and decoupled from the assertions themselves :)

Reply:
Doing the key assertion won't catch bugs where the keys are actually wrong in the first place though... many occurrences of such bugs already.